### PR TITLE
[IPEHR-183] API Get doctor info by code + integration test

### PR DESCRIPTION
**Pull request by 1db1**

### CODE REVIEW

File: src/pkg/api/user.go [hunk 1]
```diff
@@ -294,10 +292,51 @@ func (h *UserHandler) Info(c *gin.Context) {
...
+// @Param    code           path      string  true  "The pin code of the requested doctor"
+// @Success  200            {object}  model.UserInfo
+// @Failure  404            "User code is not exist"
+// @Failure  422            "The request could not be understood by the server due to incorrect syntax. The client SHOULD NOT repeat the request without modifications."
+// @Failure  500            "Is returned when an unexpected error occurs while processing a request"
+// @Router   /user/code/{code} [get]
+func (h *UserHandler) InfoByCode(c *gin.Context) {
+	codeString := c.Param("code")
+
+	codeInt, _ := strconv.Atoi(codeString)
```
Comment (Setest) [inline src/pkg/api/user.go:321]: in that case codeString can be signed (-+)

File: src/pkg/api/user.go [hunk 2]
```diff
@@ -294,10 +292,51 @@ func (h *UserHandler) Info(c *gin.Context) {
...
+	if codeInt < 10000000 || codeInt > 99999999 {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "request code must contain 8 digits"})
+		return
+	}
```
Comment (Setest) [inline src/pkg/api/user.go]: how about "00000001" i think sting more preferable with digit check

Comment (1db1) [reply src/pkg/api/user.go]: yeah, you are right, it's possible, fixed

File: src/pkg/api/user.go (hunk 1 above)
Comment (1db1) [reply src/pkg/api/user.go:321]: it is checked in the condition below

**Review by Setest — APPROVED** [review]

**Review by SealTV — APPROVED** [review]

**Pull request merged**